### PR TITLE
docs(typescript): fix typos

### PR DESCRIPTION
**Pull request by GalvinGao**
Just some typo fixes :D

#### Changelog

**New**

- `N/A`

**Changed**

- `fixed typos`

**Removed**

- `N/A`

#### Testing / Reviewing

`N/A`

### CODE REVIEW

**Comment by GalvinGao** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by andreancardona — APPROVED** [review]
thank you! :)

**Comment by aledavila** [comment]
Hi could you do a `yarn format` and commit that change please thanks :)

File: docs/guides/adding-component-types.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ the repository. There is an issue tracking curring status of this effort,
 
 - If it's not on [the storybook](https://react.carbondesignsystem.com), it's
   probably not public.
-- The entire Public API is snapshotted, you can
+- The entire Public API is snapshot-ed, you can
```
Comment (tw15egan) [inline docs/guides/adding-component-types.md]: I believe this is the correct past tense of "snapshot"

Screenshot 2023-02-23 at 2 58 51 PM

Comment (GalvinGao) [reply docs/guides/adding-component-types.md]: My bad; the spell-checking plugin I was using didn't seem to like this. Fixed.

**Review by tw15egan — APPROVED** [review]

**Pull request merged**